### DISCUSSION TITLE
Multiple instances of SDM

### DISCUSSION THREAD

**Discussion by cjaran** [gitbls/sdm#206] [General]
I'm currently working on a python wrapper of SDM that invokes sdm using subprocess, the main goal of my wrapper is to read a csv file, modify an image, then write each configuration on the burn from the csv file, for each respective device.

So i am just wondering if it would be possible to do multiple at the same time or does it need to be linear, e.g, one at a time?

**Comment by gitbls** [comment]
This sounds cool and very useful for building a fleet of disks as quickly as possible.

The only thing that you won't be able to do in parallel is *modify an image*, if they are the same physical disk IMG file.

If each parallel burn is burning from a completely separate IMG file then of course the *modify an IMG* can be done, but the good news is that you don't need to do that at all, since pretty much everything you might want to modify in the IMG can be done at burn time via plugins.

sdm-gburn does this, but sequentially, rather than in parallel. 

I'm interested in seeing how this turns out!
